src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php

```diff
@@ -12,9 +13,15 @@
      */
     private $parser;
 
-    public function __construct(Parser $parser)
+    /**
+     * @var ErrorHandler
+     */
+    private $errorHandler;
+
+    public function __construct(Parser $parser, ErrorHandler $errorHandler = null)
```

Comment:
`?ErrorHandler`?

Reply:
It's an optional `PhpParser\ErrorHandler` which is passed to the Parser, to either collect errors or throw errors.
It's set here, depending on cli options: https://github.com/maglnet/ComposerRequireChecker/pull/38/commits/dd1cb79984746ad25e0f75f076003e7373624e9a#diff-a97f42e82e15e71146e1b0d2bcf8f5c0R70

Reply:
Yes, I was suggesting not making it an optional parameter, but either a nullable one, or to provide our own `SilencingErrorHandler` or such.

Reply:
Ah, I understand, didn't know about `nullable types` until now :)
This would make future phar files incompatible with `php 7.0` since `nullable types` are new in `7.1`.
So currently I'm a bit struggling with dropping `7.0`  as long as it is supported.